src/database/schema/ChannelListSchema.ts

```diff
@@ -502,8 +503,11 @@ class ChannelList implements BaseDbSchema {
     members?: ChannelData['members'],
     anonUserInfo: AnonUserInfo | null = null
   ): ChannelList {
+    const sortedMessages = data?.messages?.sort(
+      (a, b) => moment(b.created_at).valueOf() - moment(a.created_at).valueOf()
+    );
     const isPM = channelType === 'PM';
-    const firstMessage = data?.messages[0] || data?.firstMessage;
+    const firstMessage = data?.firstMessage || sortedMessages?.[0];
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic to sort messages by creation date in descending order and then assign the first message is correct. However, please be aware that the `sort` function modifies the original array. If preserving the original order of messages elsewhere in your code is important, you might want to create a copy of the array before sorting. Also, consider error handling for invalid date formats.

```diff
- const sortedMessages = data?.messages?.sort(
-   (a, b) => moment(b.created_at).valueOf() - moment(a.created_at).valueOf()
- );
+ const sortedMessages = [...data?.messages]?.sort(
+   (a, b) => compareDesc(new Date(b.created_at), new Date(a.created_at))
+ );
const firstMessage = data?.firstMessage || sortedMessages?.[0];
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->